Matrix.java

```diff
@@ -0,0 +1,210 @@
+class Matrix
+{
+    final double[][] a;
+    
+    Matrix(double[][] A)
+    {
+	this(A,0,0);
+    }
+    Matrix(double[][] A, int m, int n)
+    {
+        int rows = A.length;
+        m = rows < m ? rows : m;
```

Comment:
Given that the description of this method is "Construct a method quickly, without checking arguments", I think we should use the matrix dimensions that are provided without checking whether they match the 2D array. If the 2D array doesn't fit, we just don't copy the parts that don't fit. If it's too small, we just leave the extra elements as 0. Thoughts?